pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux.go

```diff
@@ -109,63 +106,89 @@ func (m *Manager) Start() error {
 	if !m.isFeatureEnabled() {
 		return nil
 	}
+	stop, err := m.start()
+	if err != nil {
+		return err
+	}
+	go func() {
+		<-stop
+		time.Sleep(retryPeriod)
+		wait.Forever(func() {
+			klog.V(1).InfoS("Restart to watching the node for shutdown events")
+			stop, err := m.start()
+			if err != nil {
+				klog.ErrorS(err, "Unable to watching the node for shutdown events")
+			} else {
+				<-stop
```

Comment:
i'm a little confused here on use of `wait.Forever` + the blocking stop. Since the inner go routine will be blocking (due to waiting for stop `<-stop`) how does that interact with the retryPeriod?

My understanding is `wait.Forever` with fixed interval is used when we want to call the function periodically but I'm not sure how this works when that function is blocked... i.e. my concern is when it will be unblocked, will it be called many times for those times that it was blocked?

Reply:
Yes, after the dbus is disconnected, it will unblock here, and then try to restart.